cmd/ct_monitor/main.go

```diff
@@ -0,0 +1,167 @@
+)
+
+// This main function performs a periodic identity search.
+// Upon starting, any existing latest snapshot data is loaded and the function runs
+// indefinitely to perform identity search for every time interval that was specified.
+func main() {
+	configFilePath := flag.String("config-file", "", "path to yaml configuration file containing identity monitor settings")
+	configYamlInput := flag.String("config", "", "path to yaml configuration file containing identity monitor settings")
+	once := flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
+	serverURL := flag.String("url", publicRekorServerURL, "URL to the rekor server that is to be monitored")
+	interval := flag.Duration("interval", 5*time.Minute, "Length of interval between each periodical consistency check")
+	flag.Parse()
+
+	if *configFilePath == "" && *configYamlInput == "" {
+		log.Fatalf("empty configuration input")
+	}
+
+	if *configFilePath != "" && *configYamlInput != "" {
+		log.Fatalf("only input one of configuration file path or yaml input")
+	}
+
+	var config notifications.IdentityMonitorConfiguration
+
+	if *configFilePath != "" {
+		readConfig, err := os.ReadFile(*configFilePath)
+		if err != nil {
+			log.Fatalf("error reading from identity monitor configuration file: %v", err)
+		}
+
+		configString := string(readConfig)
+		if err := yaml.Unmarshal([]byte(configString), &config); err != nil {
+			log.Fatalf("error parsing identities: %v", err)
+		}
+	}
+
+	if *configYamlInput != "" {
+		if err := yaml.Unmarshal([]byte(*configYamlInput), &config); err != nil {
+			log.Fatalf("error parsing identities: %v", err)
+		}
+	}
+
+	var fulcioClient *ctclient.LogClient
+	fulcioClient, err := ctclient.New(*serverURL, http.DefaultClient, jsonclient.Options{})
+	if err != nil {
+		log.Fatalf("getting Fulcio client: %v", err)
+	}
+
+	allOIDMatchers, err := config.MonitoredValues.OIDMatchers.RenderOIDMatchers()
+	if err != nil {
+		fmt.Printf("error parsing OID matchers: %v", err)
+	}
+
+	monitoredValues := identity.MonitoredValues{
+		CertificateIdentities: config.MonitoredValues.CertificateIdentities,
+		OIDMatchers:           allOIDMatchers,
+	}
+
+	for _, certID := range monitoredValues.CertificateIdentities {
+		if len(certID.Issuers) == 0 {
+			fmt.Printf("Monitoring certificate subject %s\n", certID.CertSubject)
+		} else {
+			fmt.Printf("Monitoring certificate subject %s for issuer(s) %s\n", certID.CertSubject, strings.Join(certID.Issuers, ","))
+		}
+	}
+	for _, fp := range monitoredValues.Fingerprints {
+		fmt.Printf("Monitoring fingerprint %s\n", fp)
+	}
+	for _, sub := range monitoredValues.Subjects {
+		fmt.Printf("Monitoring subject %s\n", sub)
+	}
+
+	ticker := time.NewTicker(*interval)
+	defer ticker.Stop()
+
+	// To get an immediate first tick
+	for ; ; <-ticker.C {
+		inputEndIndex := config.EndIndex
+
+		var currentSTH *ctgo.SignedTreeHead
+		currentSTH, err = ct.RunConsistencyCheck(fulcioClient, config.LogInfoFile)
```

Comment:
Can we copy over the changes that `RunConsistencyCheck` returns the previous, and it's read in below? Otherwise we'll hit the same bug that we read in the latest as the previous.